File: TradeItIosEmsApi/TradeItBroker.m
#import "TradeItBroker.h"

@implementation TradeItBroker

- (NSString *)description {
    return [NSString stringWithFormat: @"[TradeItBroker: shortName=%@, longName=%@, brokerInstruments=%@]",
            self.shortName,
            self.longName,
            self.brokerInstruments];
}

// Add backwards compatible getters :(
- (NSString *)brokerShortName {
    return self.shortName;
}

- (NSString *)brokerLongName {
    return self.longName;
}

@end

File: TradeItIosTicketSDK2/TradeItBroker.swift
@objc public class TradeItBroker: NSObject, Codable {
    var shortName: String?
    var longName: String?
    var brokerInstruments: [TradeItBrokerInstrument]?
    var brokerShortName: String? {
        return shortName
    }
    var brokerLongName: String? {
        return longName
    }
    var logos: [TradeItBrokerLogo]?
    
    private enum CodingKeys: String, CodingKey {
        case shortName
        case longName
        case brokerInstruments
        case logos
    }
}
